Recipes/Recipes/MainViewController.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  MainViewController.swift
+//  Recipes
+//
+//  Created by Isaac Lyons on 9/30/19.
+//  Copyright © 2019 Lambda Inc. All rights reserved.
+//
+
+import UIKit
+
+class MainViewController: UIViewController {
+
+    @IBOutlet weak var textField: UITextField!
+    
+    var networkClient = RecipesNetworkClient()
+    var allRecipes: [Recipe] = [] {
+        didSet {
+            filterRecipes()
+        }
+    }
+    var recipesTableViewController: RecipesTableViewController? {
+        didSet{
+            recipesTableViewController?.recipes = filteredRecipes
+        }
+    }
+    var filteredRecipes: [Recipe] = [] {
+        didSet {
+            recipesTableViewController?.recipes = filteredRecipes
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        networkClient.fetchRecipes { (recipes, error) in
+            if let error = error {
+                NSLog("\(error)")
+                return
+            } else if let recipes = recipes {
+                self.allRecipes = recipes
```

Comment:
Notice that on line 40 you are setting the allRecipes  in a Network call. If you move perform this on the main thread you will notice the purple errors will go away.
```
 DispatchQueue.main.async {
                self.allRecipes = recipes
                }
```

Reply:
Fixed in aeb8c51200805b78eae782deb4f7eb08aae6acaa